lib/t/matches.ex

```diff
@@ -1183,19 +1189,27 @@ defmodule T.Matches do
   end
 
   defp matches_with_undying_events_q(query \\ named_match_q()) do
-    join(
-      query,
-      :left_lateral,
-      [m, e],
-      c in subquery(
-        MatchEvent
-        |> where(match_id: parent_as(:match).id)
-        |> where([e], e.event == "call_start" or e.event == "meeting_report")
-        |> select([e], e.timestamp)
-        |> limit(1)
-      ),
-      as: :undying_event
-    )
+    undying_events_q =
+      MatchEvent
+      |> where(match_id: parent_as(:match).id)
+      |> where([e], e.event == "call_start" or e.event == "meeting_report")
+      |> select([e], e.timestamp)
+      |> limit(1)
+
+    join(query, :left_lateral, [m], e in subquery(undying_events_q), as: :undying_event)
+  end
+
+  # counts how many users left voicemail per match
+  # 0 -> nobody
+  # 1 -> only one user in a match
+  # 2 -> both users exchanged voicemail
+  defp matches_with_voicemail_participants_count_q(query \\ named_match_q()) do
+    voicemail_count_q =
+      Voicemail
+      |> group_by([v], v.match_id)
+      |> select([v], count(v.caller_id, :distinct))
+
+    join(query, :left, [m], v in subquery(voicemail_count_q), on: v.match_id == m.id)
```

Comment:
@akhrail1996 I think I'll use this for now instead of match_events for voicemail timer. I'll join on both "undying_events" and this count and it should provide enough info to find matches to expire:

1. has undying event -> no expire
2. no undying event and has callers count = 2 -> expire if match's seven days old(er)
3. no undying event and callers count < 2 -> expire if match's two days old(er)

Reply:
no longer relevant